### PR TITLE
Upgrade packages to latest versions

### PR DESCRIPTION
**Pull request by jamwaffles**
Thank you for helping out with embedded-graphics development! Please:

- [x] Check that you've added passing tests and documentation
- [x] Add a `CHANGELOG.md` entry in the **Unreleased** section under the appropriate heading (**Added**, **Fixed**, etc) if your changes affect the **public API**
- [x] Run `rustfmt` on the project
- [x] Run `just build` (Linux/macOS only) and make sure it passes. If you use Windows, check that CI passes once you've opened the PR.

## PR description

Upgrade package versions to their latest at time of writing. Pleasantly few changes required.

### CODE REVIEW

**Review by rfuest — APPROVED** [review]
I did sneak in #662 before this PR, so that it needs to be rebased, but it looks good otherwise.

File: CHANGELOG.md [hunk 1]
```diff
@@ -41,6 +41,7 @@ Embedded Graphics is a `no_std` library for adding graphics features to display
   - `Styled::new`
 - [#651](https://github.com/embedded-graphics/embedded-graphics/pull/651), [#652](https://github.com/embedded-graphics/embedded-graphics/pull/652) Improved performance of color conversions.
 - [#662](https://github.com/embedded-graphics/embedded-graphics/pull/662) `ImageRaw::new` no longer panics if `width == 0`.
+  - **(breaking)** [#663](https://github.com/embedded-graphics/embedded-graphics/pull/663) Upgraded Cargo dependencies to their latest versions.
```
Comment (rfuest) [inline CHANGELOG.md:44]:
```suggestion
- **(breaking)** [#663](https://github.com/embedded-graphics/embedded-graphics/pull/663) Upgraded Cargo dependencies to their latest versions.
```

**Review by rfuest — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (jamwaffles) [reply CHANGELOG.md]: This is what I get for doing "save without formatting" in VSCode!

File: Cargo.toml [hunk 2]
```diff
@@ -27,15 +27,15 @@ all-features = true
 circle-ci = { repository = "embedded-graphics/embedded-graphics", branch = "master" }
 
 [dependencies]
-az = "1.1"
-fixed = { version = "0.5.7", optional = true, default-features = false }
-float-cmp = "0.8.0"
-micromath = { version = "1.1.0", default-features = false }
+az = "1.2.0"
+fixed = { version = "1.14.0", optional = true, default-features = false }
```
Comment (jamwaffles) [inline Cargo.toml:31]: To note, `fixed` v2 (currently alpha 😞) will make a big pile of its functions `const` according to the [changelog](https://gitlab.com/tspiteri/fixed/-/blob/master/RELEASES.md).

Comment (rfuest) [reply Cargo.toml:31]: That sounds great. Should we open an issue to remind ourself to update to version 2 if it is released?

**Review by rfuest — APPROVED** [review]

File: Cargo.toml (hunk 2 above)
Comment (jamwaffles) [reply Cargo.toml:31]: Opened #665 to track.

**Pull request merged**